src/chericat.c

```diff
@@ -84,6 +93,20 @@ static struct option long_options[] =
 	{0,0,0,0}
 };
 
+void handle_terminate(int sig)
```

Comment:
This seems to be basically the same as the exit path from `main`. Maybe give it a more general name and call it from there to?

Reply:
fixed.